packages/ethereum-storage/src/ethereum-storage-ethers.ts

```diff
@@ -22,59 +24,26 @@ export type StorageEventEmitter = TypedEmitter<{
 export class EthereumStorageEthers implements StorageTypes.IStorageWrite {
   private readonly logger: LogTypes.ILogger;
   private readonly ipfsStorage: StorageTypes.IIpfsStorage;
-  private readonly hashSubmitter: RequestOpenHashSubmitter;
+
   private readonly network: string;
-  private readonly provider: providers.JsonRpcProvider;
-  private enableEip1559 = true;
+  private readonly txSubmitter: EthereumTransactionSubmitter;
 
   constructor({ network, signer, ipfsStorage, logger }: StorageProps) {
     this.logger = logger || new Utils.SimpleLogger();
     this.ipfsStorage = ipfsStorage;
     this.network = network;
-    this.provider = signer.provider as providers.JsonRpcProvider;
-    this.hashSubmitter = requestHashSubmitterArtifact.connect(
-      network,
-      signer,
-    ) as RequestOpenHashSubmitter; // type mismatch with ethers.
+    this.txSubmitter = new EthereumTransactionSubmitter({ network, signer, logger });
```

Comment:
refactoring to isolate the part where we submit the transaction to the blockchain, for better reusability.